stock_no_negative/tests/test_stock_no_negative.py

```diff
@@ -165,17 +165,27 @@ def test_true_allow_negative_stock_product_with_lot(self):
         """Assert that negative stock levels are allowed when
         the allow_negative_stock is set active in the product with lot"""
         self.product_with_lot.allow_negative_stock = True
+        quant_rec = self.env["stock.quant"].create(
+            {
+                "product_id": self.product_with_lot.id,
+                "location_id": self.location_id.id,
+                "quantity": 100,
+                "lot_id": self.lot1.id,
+            }
+        )
         self.stock_picking_with_lot.action_confirm()
-        self.stock_picking_with_lot.move_ids.quantity_done = 100
+        self.stock_picking_with_lot.move_ids.quantity = 200
         with self.assertRaises(UserError):
+            self.stock_picking_with_lot.move_ids.picked = True
             self.stock_picking_with_lot._action_done()
-        self.stock_picking_with_lot.move_ids.move_line_ids[0].lot_id = self.lot1.id
+        self.stock_picking_with_lot.move_ids.move_line_ids[0].write(
+            {"quant_id": quant_rec.id, "lot_id": self.lot1.id}
+        )
         self.stock_picking_with_lot._action_done()
         quant = self.env["stock.quant"].search(
             [
                 ("product_id", "=", self.product_with_lot.id),
-                ("location_id", "=", self.location_id.id),
                 ("lot_id", "=", self.lot1.id),
             ]
         )
-        self.assertEqual(quant.quantity, -100)
+        self.assertEqual(quant.quantity, 100)
```

Comment:
@fd-oerp Why changing this as this test is checking negative is allowed ?

Reply:
Hi @rousseldenis , the assert giving the positive quantity, I have also updated some code in this method

Reply:
Be careful. During migration, the results of test should - generally - be kept.

As the test is reflecting the purpose of the code. Reading the comment at the beginning of the test function would have oriented you through a questioning.

So two questions remain:

- Is this module do not behave as expected?
- Has Odoo changed core module to avoid negative quantities?

Both should be answered.

Reply:
Hi @rousseldenis ,
Thanks for clarity, 1) Yes this module works as expected 2)No, by using this module we can allow to have a negative quantity for product with lot as well.

Reply:
> Hi @celm1990 @rousseldenis any update on this PR, anything needs to be updated from my side?

It is not clear to me because you changed the tests, same doubt as @rousseldenis.

Reply:
Hi @celm1990 

I just updated that test case with the necessary changes, please review and proceed or let me know if anything needs to be done from my side